Store userTheme preference in localStorage

ivinable2000 opened the pull request:
Updated the toggleMode function to update user theme in local Storage, and added snippet of code to remember theme from local Storage
closes #20

H-Richard commented on src/App.tsx:
```diff
@@ -14,10 +14,24 @@ import Footer from './components/Footer';
   }, [themeType, setThemeType]);
 
   const theme: Theme = useMemo(() => {
+    if (!localStorage.getItem('themeType')) {
+      if (themeType === 'light') {
+        localStorage.setItem('themeType', 'dark');
+      } else {
+        localStorage.setItem('themeType', 'light');
+      }
+    } else {
+      setThemeType(localStorage.getItem('themeType') === 'light' ? 'light' : 'dark');
+    }
```
This should be in a `useEffect` where if the localStorage theme isn't set we set it to default. Also is there any way we can use eventlisteners to simplify some of this code? we can listen for the 'storage' event with `window.addEventListener('storage', ...)` and we check if the localstorage had any changes to do with theme in the 2nd arg

ivinable2000 commented:
I added a useEffect function to set localStorage theme if it doesnt exist. 

I also added an event listener but it is only called when the storage is changed by a different window, it does not respond to storage updates from the same window (So if you make a change in one window you will see it in the second one but not the original). 
https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event

So to see the event handler (setThemeType) must be called manually to make changes in same page.
